### PR TITLE
Remove code deprecated in 2.6.0 

### LINKED ISSUE #7106: Remove code deprecated in 2.6.0
This is a handle turning exercise - no coding skill is needed, you just need to be thorough :)

1. Find out what needs to be removed: `grep -R deprec zap/src/main/java | grep "2.6.0"` - there should be 41 matches
2. Remove the classes / methods / fields found above
3. Add a `ZAP` header to any `paros` code changed (see examples below)
4. Add the affected classes / methods / fields to `zap/zap.gradle.kts` (see below)
5. Run `spotlessApply`
6. Run `./gradlew japicmp` - it should pass, if not check the report it generates
7. Update the corresponding help file (see below)

Example PRs:

2.2.0 + 2.3.0: code: https://github.com/zaproxy/zaproxy/pull/7067 help: https://github.com/zaproxy/zap-core-help/pull/414
2.4.x: code: https://github.com/zaproxy/zaproxy/pull/7068 help: https://github.com/zaproxy/zap-core-help/pull/415
2.5.0: code: https://github.com/zaproxy/zaproxy/pull/7102 help: https://github.com/zaproxy/zap-core-help/pull/420

### PR DESCRIPTION
**Pull request by jini0x**
Fix #7106.

Signed-off-by: Hichem Yagoubi <hichem.yagoubi1@gmail.com>

### CODE REVIEW

**Comment by jini0x** [comment]
3 failing... is that a problem?

**Comment by thc202** [comment]
No tests should be failing, in this case you need to update them since the constructor no longer exists.

**Comment by jini0x** [comment]
So, should I update the tests?

**Comment by jini0x** [comment]
Hi, I need more help here please, I changed:
boolean isAddOn(String fileName) -> boolean isAddOnFileName(String fileName)

boolean isAddOn(File f) -> boolean isAddOn(Path file) 

AddOn(String fileName) or AddOn(File file) ->  AddOn(Path file)

for example: 
   AddOn addOnA1 = new AddOn("test-alpha-1.zap"); to :
   AddOn addOnA1 = new AddOn(Files.createDirectory(newTempDir().resolve("test-alpha-1.zap")));


image

File: zap/src/main/java/org/zaproxy/zap/extension/brk/PopupMenuRemove.java [hunk 1]
```diff
@@ -46,11 +46,11 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
...
-            if (extension.canRemoveBreakpoint()) {
-                this.setEnabled(true);
-            } else {
-                this.setEnabled(false);
-            }
+            //            if (extension.canRemoveBreakpoint()) {
+            //                this.setEnabled(true);
+            //            } else {
+            //                this.setEnabled(false);
+            //            }
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/brk/PopupMenuRemove.java:53]: Change to `setEnabled(true);`. (The warn suppression should be removed.)

File: zap/zap.gradle.kts [hunk 2]
```diff
@@ -166,7 +166,8 @@ val japicmp by tasks.registering(JapicmpTask::class) {
         "org.parosproxy.paros.core.scanner.Alert#WARNING",
         "org.parosproxy.paros.view.View#DISPLAY_OPTION_BOTTOM_FULL",
         "org.parosproxy.paros.view.View#DISPLAY_OPTION_LEFT_FULL",
-        "org.parosproxy.paros.view.View#DISPLAY_OPTION_TOP_FULL"
+        "org.parosproxy.paros.view.View#DISPLAY_OPTION_TOP_FULL",
+        "org.parosproxy.paros.model.HistoryReference#TYPE_RESERVED_11"
```
Comment (thc202) [inline zap/zap.gradle.kts:170]: The entries should be added in alphabetic order.

File: zap/src/main/java/org/parosproxy/paros/CommandLine.java [hunk 3]
```diff
@@ -49,6 +49,7 @@
 // ZAP: 2020/11/26 Use Log4j 2 classes for logging.
 // ZAP: 2021/05/14 Remove redundant type arguments.
 // ZAP: 2022/02/09 No longer parse host/port and deprecate related code.
+// ZAP: 2022/02/28 Removed deprecated getConfigs()
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/CommandLine.java:52]: Could just say `Remove code deprecated in 2.6.0`, same in other files.

File: zap/src/test/java/org/zaproxy/zap/control/AddOnUnitTest.java [hunk 4]
```diff
@@ -60,88 +60,36 @@
...
-    @Test
-    @SuppressWarnings("deprecation")
-    void testVersion() throws Exception {
-        AddOn addOn = new AddOn("test-alpha-1.zap");
-        assertThat(addOn.getVersion().toString(), is(equalTo("1.0.0")));
-        assertThat(addOn.getFileVersion(), is(1));
-    }
-
     @Test
     @SuppressWarnings("deprecation")
```
Comment (thc202) [inline zap/src/test/java/org/zaproxy/zap/control/AddOnUnitTest.java:64]: These should be removed now.

File: zap/zap.gradle.kts [hunk 5]
```diff
@@ -206,23 +219,28 @@ val japicmp by tasks.registering(JapicmpTask::class) {
...
         "org.parosproxy.paros.extension.option.OptionsParamView#getShowMainToolbar()",
         "org.parosproxy.paros.extension.option.OptionsParamView#setShowMainToolbar(int)",
+        "org.parosproxy.paros.model.Session#addGlobalExcludeURLRegexs(java.lang.String)",
+        "org.parosproxy.paros.model.Session#setGlobalExcludeURLRegexs(java.util.List)",
         "org.parosproxy.paros.network.ConnectionParam#getProxyChainSkipName()",
         "org.parosproxy.paros.network.ConnectionParam#setProxyChainSkipName(java.lang.String)",
         "org.parosproxy.paros.view.AbstractFrame#loadIconImages()",
-        "org.parosproxy.paros.view.MainFrame#changeDisplayOption(int)",
         "org.parosproxy.paros.view.MainFrame#MainFrame(int)",
+        "org.parosproxy.paros.view.MainFrame#changeDisplayOption(int)",
```
Comment (thc202) [inline zap/zap.gradle.kts:239]: Why this change? There are other changes that make them not to be in alphabetic order.

Comment (jini0x) [reply zap/zap.gradle.kts:239]: ohh I sorted them with the IDE, I will make changes manually now.

File: zap/src/test/java/org/zaproxy/zap/control/AddOnUnitTest.java (hunk 4 above)
Comment (jini0x) [reply zap/src/test/java/org/zaproxy/zap/control/AddOnUnitTest.java:64]: I'm not sure I'm following up with you sir, what should be removed?

Comment (thc202) [reply zap/src/test/java/org/zaproxy/zap/control/AddOnUnitTest.java:64]: The `SuppressWarnings` annotations, the methods should no longer have any deprecation usage.

Comment (jini0x) [reply zap/src/test/java/org/zaproxy/zap/control/AddOnUnitTest.java:64]: okkk, now I get it... thanks

File: zap/zap.gradle.kts (hunk 5 above)
Comment (jini0x) [reply zap/zap.gradle.kts:239]: the sort was case sensitive, everything is cool now.

File: zap/src/main/java/org/parosproxy/paros/CommandLine.java (hunk 3 above)
Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/CommandLine.java:52]: This was marked as resolved but not code change or following up comment.

**Comment by thc202** [comment]
Could you fix up the commits?

**Comment by jini0x** [comment]
Is it normal that the build lasts too long?

**Comment by kingthorin** [comment]
It was waiting to be approved because you're a first time contributor. I'll kick it off.

File: zap/src/main/java/org/parosproxy/paros/CommandLine.java (hunk 3 above)
Comment (jini0x) [reply zap/src/main/java/org/parosproxy/paros/CommandLine.java:52]: oh I'm sorry I thought it was just a suggestion for future contribution because I took this template from Simon old pull req

Comment (thc202) [reply zap/src/main/java/org/parosproxy/paros/CommandLine.java:52]: It is but I'd be better if done mainly to avoid the multiline comments.

Comment (jini0x) [reply zap/src/main/java/org/parosproxy/paros/CommandLine.java]: Done!

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Comment by jini0x** [comment]
@thc202 Thanks for your help
Hichem Yagoubi ([@jini0x](https://github.com/jini0x))
